keras_core/backend/tensorflow/nn.py

```diff
@@ -74,26 +74,27 @@ def gelu(x, approximate=True):
     return tf.nn.gelu(x, approximate)
 
 
-def softmax(x, axis=None):
+def softmax(x, axis=-1):
     logits = x
     if axis is None:
         # Unlike numpy, tf will handle axis=None as axis=-1.
         # We need this workaround for the reduction on every dim.
-        logits_exp = tf.exp(logits)
-        output = logits_exp / tf.reduce_sum(logits_exp, keepdims=True)
+        output = tf.reshape(x, [-1])
+        output = tf.nn.softmax(output, axis=-1)
+        output = tf.reshape(output, tf.shape(x))
     else:
         output = tf.nn.softmax(x, axis=axis)
     output._keras_logits = logits
```

Comment:
why is this needed btw?

Reply:
When computing crossentropy, we try to do it from logits rather from softmax/sigmoid values.